web/src/components/storage/utils.js

```diff
@@ -215,7 +215,7 @@ const deviceLabel = (device) => {
 const deviceChildren = (device) => {
   const partitionTableChildren = (partitionTable) => {
     const { partitions, unusedSlots } = partitionTable;
-    const children = partitions.concat(unusedSlots);
+    const children = partitions.concat(unusedSlots).filter(i => !!i);
```

Comment:
I detected that sometimes this ends up with [partition, partition, undefined]. We should add a test for this util.